ivadomed/utils.py

```diff
@@ -260,8 +261,7 @@ def structurewise_uncertainty(fname_lst, fname_hard, fname_unc_vox, fname_out):
     elif len(data_l_lst.shape) == 3:
         data_l_lst = np.transpose(data_l_lst, (0, 2, 3, 1))
     else:
-        print("object does not appear to be 2D or 3D, please check the network output")
-        exit(1)
+        raise Exception("object does not appear to be 2D or 3D, please check input")
```

Comment:
there is probably more specific exception to use, like ValueError? @Drulex ?
also: the else is confusing as it checks for dim=4 or 3, and the message tells to input dim=2 or 3
maybe you can do the check before?  for better code clarity

Reply:
Yes `ValueError` is appropriate. @lrouhier see usage of built-in exceptions here https://docs.python.org/3/library/exceptions.html#ValueError

Also make the message more clear, for example: `raise ValueError("Input data of shape {} is not valid. Shape must be 3 or 4".format(data_l_lst.shape))`

Reply:
done in 65037df. Thanks for the tips @Drulex ! (the link is not obvious in https://docs.python.org/3/tutorial/errors.html)

Reply:
> the else is confusing as it checks for dim=4 or 3, and the message tells to input dim=2 or 3
>maybe you can do the check before? for better code clarity

Basically I am doing this because the 4th dimension is the channel, but your input is not a commonly called "4D image", it is a 3D image with a channel dimension. The dimension will always be there but it's not what is usually considered a 4D image (e.g., it's not DTI or something like that). I thought that it would be less confusing for end user (that may not be used to considering channels as a dimension). I can change that.

Reply:
Changed for right numbers in ad094bc